contracts/allocators/OnsenAllocatorV2.sol

```diff
@@ -0,0 +1,268 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.10;
+import "../libraries/Address.sol";
+
+///  ========== TYPES ==========
+import "../types/BaseAllocator.sol";
+
+///  ========== INTERFACES ==========
+import "./interfaces/IMasterChef.sol";
+import "./interfaces/ISushiBar.sol";
+
+struct PoolData {
+    bool poolActive;
+    uint256 id;
+}
+
+/**
+ *  @notice Contract deploys liquidity from treasury into the Onsen program,
+ *  earning $SUSHI that can be staked and/or deposited into the treasury.
+ */
+contract OnsenAllocatorV2 is BaseAllocator {
+    using SafeERC20 for IERC20;
+
+    /// ========== STATE VARIABLES ==========
+
+    address public sushi; /// $SUSHI token
+    address public xSushi; /// $xSUSHI token
+    address public masterChef; /// Onsen contract
+
+    address public treasury; /// Olympus Treasury
+
+    mapping(address => PoolData) internal _lpToOnsenId;
+
+    /// ========== CONSTRUCTOR ==========
+
+    constructor(
+        address _chef,
+        address _sushi,
+        address _xSushi,
+        address _treasury,
+        AllocatorInitData memory data
```

Comment:
What if we want to add more tokens? Add a setter for more LP tokens, should be easy.

Reply:
@PARABRAHMAN0  Sorry but not sure about what this means, or maybe is already added based on a previous comment with the following 2 functions?
function setLPTokenOnsenPoolId(address lpToken, uint256 onsenPoolId) external onlyGuardian
function getLPTokenOnsenPoolId(address lpToken) external view returns (bool, uint256)